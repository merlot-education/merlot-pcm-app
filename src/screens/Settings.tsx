import { StackScreenProps } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, View, Alert } from 'react-native'
import { getVersion, getBuildNumber } from 'react-native-device-info'
import { borderRadius, Colors, TextTheme } from '../theme/theme'
import { Screens, SettingStackParams } from '../types/navigators'
import { SettingListItem, Text } from '../components'

type SettingsProps = {
  navigation: StackScreenProps<SettingStackParams, Screens.Settings>
  setAuthenticated: React.Dispatch<React.SetStateAction<boolean>>
}
const styles = StyleSheet.create({
  container: {
    width: '100%',
    padding: 20,
  },
  groupHeader: {
    ...TextTheme.normal,
    marginBottom: 8,
  },
  bodyText: {
    ...TextTheme.normal,
    flexShrink: 1,
    color: Colors.primary,
  },
  rowGroup: {
    borderRadius: borderRadius * 2,
    backgroundColor: Colors.primaryLight,
    marginBottom: 10,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 12,
  },
})

const Settings: React.FC<SettingsProps> = ({
  navigation,
  setAuthenticated,
}) => {
  const { t } = useTranslation()
  const logoff = () =>
    Alert.alert(t('Settings.Logout'), t('Settings.LogoutMsg'), [
      {
        text: t('Settings.Yes'),
        onPress: () => setAuthenticated(false),
      },
      { text: t('Settings.No') },
    ])

  return (
    <View style={styles.container}>
      <Text style={styles.groupHeader}>{t('Settings.AppPreferences')}</Text>
      <SettingListItem
        title={t('Settings.ChangePin')}
        onPress={() => navigation.navigate(Screens.ChangePin)}
      />
      <SettingListItem
        title={t('Settings.Language')}
        onPress={() => navigation.navigate(Screens.Language)}
      />
      <SettingListItem title={t('Settings.Logout')} onPress={logoff} />
      <Text style={styles.groupHeader}>{t('Settings.AboutApp')}</Text>
      <View style={styles.rowGroup}>
        <View style={styles.row}>
          <Text style={styles.bodyText}>{t('Settings.Version')}</Text>
          <Text
            style={styles.bodyText}
          >{`${getVersion()}.${getBuildNumber()}`}</Text>
        </View>
      </View>
    </View>
  )
}

export default Settings
