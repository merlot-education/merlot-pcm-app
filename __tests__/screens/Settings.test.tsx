import React from 'react'
import { create } from 'react-test-renderer'
import Settings from '../../src/screens/Settings'

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('react-native-device-info', () => {
  return {
    getVersion: () => 1,
    getBuildNumber: () => 1,
  }
})

describe('Home Screen', () => {
  it('Renders correctly', () => {
    const tree = create(<Settings />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
