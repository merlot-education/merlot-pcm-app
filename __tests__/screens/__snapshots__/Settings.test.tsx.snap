// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1E9BF2",
        },
        Object {
          "color": "#1E9BF2",
          "fontSize": 18,
          "fontWeight": "normal",
          "marginBottom": 8,
        },
      ]
    }
  >
    Settings.AppPreferences
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#CCE8F4",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
          },
          Object {
            "color": "#1E9BF2",
            "flexShrink": 1,
            "fontSize": 18,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Settings.ChangePin
    </Text>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#CCE8F4",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
          },
          Object {
            "color": "#1E9BF2",
            "flexShrink": 1,
            "fontSize": 18,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Settings.Language
    </Text>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#1E9BF2",
        },
        Object {
          "color": "#1E9BF2",
          "fontSize": 18,
          "fontWeight": "normal",
          "marginBottom": 8,
        },
      ]
    }
  >
    Settings.AboutApp
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#CCE8F4",
        "borderRadius": 8,
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1E9BF2",
            },
            Object {
              "color": "#1E9BF2",
              "flexShrink": 1,
              "fontSize": 18,
              "fontWeight": "normal",
            },
          ]
        }
      >
        Settings.Version
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1E9BF2",
            },
            Object {
              "color": "#1E9BF2",
              "flexShrink": 1,
              "fontSize": 18,
              "fontWeight": "normal",
            },
          ]
        }
      >
        1-1
      </Text>
    </View>
  </View>
</View>
`;
